acme/authority.go

```diff
@@ -263,21 +265,37 @@ func (a *Authority) ValidateChallenge(p provisioner.Interface, accID, chID strin
 	if accID != ch.getAccountID() {
 		return nil, UnauthorizedErr(errors.New("account does not own challenge"))
 	}
+	retry := ch.getRetry()
+	if retry.Active {
+		return ch.toACME(a.db, a.dir, p)
+	}
+	retry.Mux.Lock()
```

Comment:
Not sure we need a mutex. I can't think of why multiple routines would be trying to write retry at once. On the read side I assume writes to the db are atomic and reads won't view partially updated data. TODO: verify. cc @dopey